src/app/modules/shared/services/shopping.cart.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { OrderProduct } from '../types/product.order';
+
+@Injectable(
+ {providedIn:'root'}
+)
+export class ShoppingCartService {
+  private shoppingCart: OrderProduct[] = [];
+  public getShoppingCart(){
+    return this.shoppingCart;
+  }
+
+  public addToShoppingCart(product: OrderProduct): OrderProduct[] {
+    const foundProduct = this.shoppingCart.find((p) => p.id === product.id);
+    if (!foundProduct) {
+      product.quantity=1;
+      this.shoppingCart.push(product);
+    } else {
+      this.shoppingCart.find(p=>{
+        if (p.id===product.id)
```

Comment:
Get used to put brackets even though there is only one statement. So you should have something like 
``` typescript
if( ...){
    ...
} else {
   ...
}